orm/helpers.go

```diff
@@ -0,0 +1,70 @@
+package orm
+
+import (
+	"reflect"
+
+	"github.com/iov-one/weave/errors"
+)
+
+type limitedIterator struct {
+	// n is count of remaining elements
+	n int
+	// iterator is the underlying iterator
+	iter SerialModelIterator
+}
+
+var _ SerialModelIterator = (*limitedIterator)(nil)
+
+func (l *limitedIterator) LoadNext(dest SerialModel) error {
+	if l.n > 0 {
+		l.n--
+		return l.iter.LoadNext(dest)
+	}
+	return errors.Wrap(errors.ErrIteratorDone, "iterator limit reached")
+}
+
+func (l *limitedIterator) Release() {
+	l.iter.Release()
+}
+
+func WithLimit(iter SerialModelIterator, limit int) (SerialModelIterator, error) {
+	if limit < 1 {
+		return nil, errors.Wrap(errors.ErrInput, "invalid limit")
+	}
+	return &limitedIterator{iter: iter, n: limit}, nil
+}
+
+func ToSlice(iter SerialModelIterator, destination SerialModelSlicePtr) error {
+	dest := reflect.ValueOf(destination)
+	if dest.Kind() != reflect.Ptr {
+		return errors.Wrap(errors.ErrType, "destination must be a pointer to slice of SerialModels")
+	}
+	if dest.IsNil() {
+		return errors.Wrap(errors.ErrImmutable, "got nil pointer")
+	}
+	dest = dest.Elem()
+	if dest.Kind() != reflect.Slice {
+		return errors.Wrap(errors.ErrType, "destination must be a pointer to slice of SerialModels")
+	}
+
+	// Allocate memory depending on the type of the destination
```

Comment:
This comment might be misleading, because below code does not allocate memory. It only determines the type of the model.